java-checks/src/main/java/org/sonar/java/filters/LombokFilter.java

```diff
@@ -117,11 +124,16 @@ public void visitClass(ClassTree tree) {
     excludeLinesIfTrue(generatesEquals, tree, EqualsNotOverriddenInSubclassCheck.class, EqualsNotOverridenWithCompareToCheck.class);
     excludeLinesIfTrue(generatesPrivateConstructor(tree), tree, UtilityClassWithPublicConstructorCheck.class);
     excludeLinesIfTrue(usesAnnotation(tree, UTILITY_CLASS), tree, BadFieldNameCheck.class, ConstantsShouldBeStaticFinalCheck.class);
-    excludeLinesIfTrue(usesAnnotation(tree, Collections.singletonList("lombok.Value")), tree, FieldModifierCheck.class);
-
     super.visitClass(tree);
   }
 
+  @Override
+  public void visitVariable(VariableTree tree) {
```

Comment:
We will go through this code for every variable declaration.
Why not going for a top-down approach: starting from a Class, and going down only when you find a relevant annotation?

Reply:
Good point. I have updated the code to only scan variables for classes with relevant annotations